readthedocs/core/resolver.py

```diff
@@ -214,6 +214,58 @@ def resolve(
         )
         return urlunparse((protocol, domain, path, '', query_params, ''))
 
+    def _get_root_path(self, project):
+        """
+        Returns the root path for a project.
+
+        If the project is a subproject, it will return ``/projects/<project-slug>/``.
+        If the project is a main project, it will return ``/``.
+        This will respect the custom urlconf of the project if it's defined.
+        """
+        custom_prefix = None
+        if project.urlconf:
+            custom_prefix = project.urlconf.split("$", 1)[0]
+
+        if project.is_subproject:
+            prefix = custom_prefix or "projects"
+            return join_url_path(prefix, project.slug, "/")
+
+        prefix = custom_prefix or "/"
+        return join_url_path(prefix, "/")
+
+    def resolve_root(self, project, external_version_slug=None):
```

Comment:
Do we need to say `full_root` or similar here? Is a `root_path` only the URL section, and `root` is the whole? Words are hard. 🤯

Reply:
Yeah, I'm using root_path to refer to the path only, and root includes the domain, these are new terms, don't think we had the need to use them before p:

Reply:
Also, instead of using `resolve_`, for this method I think it fits better the word `get_`.

Finally, it will be called `get_full_url_prefix()`, which seems more explicit to me.

Reply:
I'd use path and URL instead of URL/full_url.